src/test/java/com/google/devtools/build/lib/skylark/SkylarkDefinedAspectsTest.java

```diff
@@ -904,6 +904,39 @@ public void aspectFailingOrphanArtifacts() throws Exception {
             + "test/missing_in_action.txt\n");
   }
 
+  @Test
+  public void aspectSkippingOrphanArtifactsWithLocation() throws Exception {
+    scratch.file("aspects/BUILD", "");
+    scratch.file(
+        "aspects/print.bzl",
+        "def _print_expanded_copts_impl(target, ctx):",
+        "    # Make sure the rule has a srcs attribute.",
+        "    if hasattr(ctx.rule.attr, 'cmd'):",
+        "        cmd = ctx.rule.attr.cmd",
+        "        print(\"{} -> {}\".format(cmd, ctx.expand_location(cmd, [])))",
+        "    return []",
+        "",
+        "print_expanded_copts = aspect(",
+        "    implementation = _print_expanded_copts_impl,",
+        "    attr_aspects = ['deps'],",
```

Comment:
Why do you need to propagate this aspect anywhere? Why not just remove this line and be content with the aspect only being applied to the genrule?

Reply:
Ditto